google-api-client/src/main/java/com/google/api/client/googleapis/mtls/MtlsUtils.java

```diff
@@ -81,33 +81,26 @@ public String getKeyStorePassword() {
 
     @Override
     public KeyStore getKeyStore() throws IOException, GeneralSecurityException {
-      // Load the cert provider command from the json file.
-      InputStream stream;
       try {
-        stream = new FileInputStream(metadataPath);
+        // Load the cert provider command from the json file.
+        InputStream stream = new FileInputStream(metadataPath);
+        List<String> command = extractCertificateProviderCommand(stream);
+
+        // Run the command and timeout after 1000 milliseconds.
+        Process process = new ProcessBuilder(command).start();
+        int exitCode = runCertificateProviderCommand(process, 1000);
+        if (exitCode != 0) {
+          throw new IOException(String.format("Failed to execute cert provider command with exit code: %d", exitCode));
```

Comment:
simple string concatenation would be clearer

Reply:
done